### ISSUE TITLE
Conflicting fields of "Make payment" claim

### ISSUE THREAD

**Issue by proshin-roman** [zerocracy/farm#1162]
If you look at [make_payment.groovy](https://github.com/zerocracy/farm/blob/master/src/main/groovy/com/zerocracy/stk/pm/cost/make_payment.groovy) you will see that claim "Make payment" may have fields "cash" and "minutes". But the logic of that stakeholder first checks if "cash" exists and uses it ignoring "minutes" field. IMO it's error prone, as claims creators might ignore this logic and then it will produce wrong behavior.

How it should be:
- claim "Make payment" may contain either "cash" or "minutes" field, but not both.
- other related claims (which are might produce "Make payment" claim) should also be refactored.

If both those fields are really required by some business logic, we can split "Make payment" claim to two different so that it's clear how it works.

**Comment by 0crat** [comment]
@g4s8[/z](https://www.0crat.com/u/g4s8) please, pay attention to this issue

**Comment by g4s8** [comment]
@proshin-roman I agree that it's not clear. I think we should separate payment claim from reputation claims, so "Make payment" stakeholders will handle only payment requests (cash and shares) and reputation stakeholders only points. 
@yegor256 what do you think?
Also this bug is related: #1281

**Comment by yegor256** [comment]
@g4s8 yes, I agree

**Comment by 0crat** [comment]
Bug was reported, see [§29](http://www.zerocracy.com/policy.html#29): +15 point(s) just awarded to @proshin-roman[/z](https://www.0crat.com/u/proshin-roman)

**Comment by 0crat** [comment]
The job [#1162](https://github.com/zerocracy/farm/issues/1162) assigned to @victornoel[/z](https://www.0crat.com/u/victornoel), here is [why](https://www.0crat.com/footprint/C3NDPUA8L/2660237322); the budget is 30 minutes, see [§4](http://www.zerocracy.com/policy.html#4); please, read [§8](http://www.zerocracy.com/policy.html#8) and [§9](http://www.zerocracy.com/policy.html#9); if the task is not clear, read [this](http://www.yegor256.com/2015/02/16/it-is-not-a-school.html) and [this](http://www.yegor256.com/2015/01/15/how-to-cut-corners.html); there will be a monetary reward for this job

**Comment by 0crat** [comment]
> @0crat waiting for reporter and ARC answer ([here](https://github.com/zerocracy/farm/issues/1162#issuecomment-423803194))

@victornoel The impediment for [#1162](https://github.com/zerocracy/farm/issues/1162) was registered successfully by @victornoel[/z](https://www.0crat.com/u/victornoel)

**Comment by g4s8** [comment]
@

**Comment by 0crat** [comment]
@g4s8[/z](https://www.0crat.com/u/g4s8) please, pay attention to this issue

**Comment by g4s8** [comment]
@victornoel I don't think this solution will work, because we can't use only minutes to make a payment. Let me explain you how payment for closed job is performed:
when 0crat assigns a job to performer it calculates `estimate` for this job, it is amount of cash in `$` which will be sent to performer on close. Then, when job is closed, 0crat read this estimate value for the job and post `Make payment` claim with this cash value and fixed minutes value. Even if performer's rate changes, the estimate will be the same.
This is why we can't use centralized payment *controller* which can decide how much to pay taking into account only minutes. So I'd suggest to start with removing awards logic from `make_payment` stakeholder, and don't listen `Make payment` claim in `add_award_points` stakeholder, and use direct claim `Add award points` (which already exists) directly where needed insted of `Make payment` with empty cash value or negative value.

**Comment by 0crat** [comment]
GitHub issue is still open, won't close

**Comment by victornoel** [comment]
@g4s8 ok, thanks for the explanation, I will do that

**Comment by victornoel** [comment]
@g4s8 @proshin-roman for the moment I am blocked by #1738

**Comment by victornoel** [comment]
@proshin-roman @g4s8 PR with fix: #1740

**Comment by victornoel** [comment]
@proshin-roman #1740 has been merged and a todo has been added for continuing the work. Can we close this issue?

**Comment by 0crat** [comment]
There is an unrecoverable failure on my side. Please, submit it [here](https://github.com/zerocracy/farm/issues):

```
PID: 4@1b1e254c-29d1-4606-9ab7-cf474cb31fbd, thread: AsyncProc-6
com.jcabi.xml.StrictXML[124] java.lang.IllegalArgumentException: 1 error(s) in XML document: -1:-1: Duplicate unique value [gh:zerocracy/farm#1162] declared for identity constraint of element "verbosity".
```

[0.35.3](https://github.com/zerocracy/farm): CID: [2780741140](https://www.0crat.com/C3NDPUA8L/2780741140), Type: "Close job", Author: "proshin-roman"

**Comment by victornoel** [comment]
@0crat status

**Comment by 0crat** [comment]
@ypshenychka[/z](https://www.0crat.com/u/ypshenychka) please review this job completed by @victornoel[/z](https://www.0crat.com/u/victornoel), as in [§30](http://www.zerocracy.com/policy.html#30); the job will be fully closed and all payments will be made when the quality review is completed

**Comment by 0crat** [comment]
The job [#1162](https://github.com/zerocracy/farm/issues/1162) is now out of scope

**Comment by 0crat** [comment]
> @0crat status ([here](https://github.com/zerocracy/farm/issues/1162#issuecomment-427287338))

@victornoel This is what I know about this job in [C3NDPUA8L](https://www.0crat.com/p/C3NDPUA8L), as in [§32](http://www.zerocracy.com/policy.html#32):

  * The job [#1162](https://github.com/zerocracy/farm/issues/1162) is not in scope
  * The job is waiting quality review verdict by @ypshenychka[/z](https://www.0crat.com/u/ypshenychka) for 19min
  * @victornoel[/z](https://www.0crat.com/u/victornoel) will receive 30 minutes after quality review
  * Job [footprint](https://www.0crat.com/footprint/C3NDPUA8L?q=%7Bjob%3A%27gh%3Azerocracy%2Ffarm%231162%27%2Ctype%3A%7B%24not%3A%2F%5ENotify%2F%7D%7D) (restricted area)

**Comment by ypshenychka** [comment]
@0crat quality good

**Comment by 0crat** [comment]
Order was finished, quality is "good": +35 point(s) just awarded to @victornoel[/z](https://www.0crat.com/u/victornoel)

**Comment by 0crat** [comment]
Quality review completed: +8 point(s) just awarded to @ypshenychka[/z](https://www.0crat.com/u/ypshenychka)

**Comment by 0pdd** [comment]
@proshin-roman the puzzle [#1765](https://github.com/zerocracy/farm/issues/1765) is still not solved.

**Comment by 0pdd** [comment]
@proshin-roman the puzzle [#1796](https://github.com/zerocracy/farm/issues/1796) is still not solved; solved: [#1765](https://github.com/zerocracy/farm/issues/1765).

**Comment by 0pdd** [comment]
@proshin-roman the puzzle [#unknown]() is still not solved.
